frontend/src/metabase/dashboard/components/ClickBehaviorSidebar/ClickBehaviorSidebar.tsx

```diff
@@ -158,6 +160,7 @@ export function ClickBehaviorSidebar({
     }
     setOriginalColumnVizSettings(null);
     setSelectedColumn(null);
+    setTypeSelectorVisible(true);
```

Comment:
Hi @kamilmielnik . This issue https://github.com/metabase/metabase/issues/34845 has two problems:

1. The current selected type was "Unknown".
2. The sidebar should show type selector instead of just one option.

The first issue solved by the `changedType` fixes above. The second issue solved by this line. When the user click the header, the type selector visibility isn't changed.

I've remove the TODO related to the issue above and rerun the E2E tests for the click behavior, here's what I got locally:

![CleanShot 2023-12-14 at 01 25 50@2x](https://github.com/metabase/metabase/assets/15979404/4cc1d9f7-1649-4c1e-957e-f13f98b61322)

Would love to hear your feedback!

Reply:
@ibedwi I tried it with and without this line and I didn't notice any difference.
Could you please write down the steps where it makes a difference?

Reply:
Thank you for the detailed explanation!

> By updating the isTypeSelectorVisible value to true when the user taken back to step 1 will make sure that the **user will see the type selection step again when clicking on any column**.

This behavior is only desired when editing the default "drill-through" click behavior.
:x: When editing "link" or "crossfilter" click behaviors, user should be taken to the very last step of the wizard, not the click behavior type selection.

----

To rephrase, just for clarity:

1. When user selects a column that has the default "drill-through" click behavior - they should be taken [here](https://github.com/metabase/metabase/assets/6830683/a8220771-b9d2-4845-9e25-2b871898f122).
2. When user selects a column that has either "link" or "crossfilter" click behavior - they should be taken [here](https://github.com/metabase/metabase/assets/6830683/9785b135-d23d-4b58-86d1-fd41ec2b8ca1).

----

And I think you missed my [other comment](https://github.com/metabase/metabase/pull/35863#issuecomment-1844978101) - the e2e tests need to be updated.

Reply:
> To rephrase, just for clarity:
> 
> When user selects a column that has the default "drill-through" click behavior - they should be taken [here](https://github.com/metabase/metabase/assets/6830683/a8220771-b9d2-4845-9e25-2b871898f122).
When user selects a column that has either "link" or "crossfilter" click behavior - they should be taken [here](https://github.com/metabase/metabase/assets/6830683/9785b135-d23d-4b58-86d1-fd41ec2b8ca1).

Hi @kamilmielnik , 
I apologize for my misunderstanding about the behavior. I revert the changes that  always turn `isTypeSelectorVisible` to true.

To keep showing the default ("drill-through") click behavior, I change [this line](https://github.com/metabase/metabase/pull/35863/files#diff-7111f88ba09a9ef63ac313c293e95c130e0b4c260e3eb824975fb6c43c8b9b07R69) to:

```tsx
// ClickBehaviorSidebarContent.tsx

if (isTypeSelectorVisible || finalClickBehavior.type === "actionMenu") //...
```

I think this is a correct way to show the "drill-through" click behavior since it represented by `undefined` or `ActionMenuClickBehavior` type. Since we already create a variable that represent the click behavior, `finalClickBehavior`, I just use that variable in the conditional statement.

This is the demo of the latest behavior:


https://github.com/metabase/metabase/assets/15979404/98573446-5c1f-4579-b8ed-becfc0f06590

-- 


> And I think you missed my https://github.com/metabase/metabase/pull/35863#issuecomment-1844978101 - the e2e tests need to be updated.

I've also updated the e2e tests. I think I missed something and the changes were reverted when I rebased this PR from the `master` branch.